volue_insight_timeseries/util.py

```diff
@@ -26,29 +27,38 @@
 
 # Frequency mapping from TS to Pandas
 _TS_FREQ_TABLE = {
-    'Y': 'AS',
+    'Y': 'YS',
     'S': '2QS',
     'Q': 'QS',
     'M': 'MS',
     'W': 'W-MON',
-    'H12': '12H',
-    'H6': '6H',
-    'H3': '3H',
-    'MIN30': '30T',
-    'MIN15': '15T',
-    'MIN5': '5T',
-    'MIN': 'T',
+    'H12': '12h',
+    'H6': '6h',
+    'H3': '3h',
+    'MIN30': '30min',
+    'MIN15': '15min',
+    'MIN5': '5min',
+    'MIN': 'min',
 }
-# Mapping from Pandas to TS is built from map above, with some additions
+
+# Mapping from Pandas to TS is built from map above,
+# with some additions to support older versions of pandas
 _PANDAS_FREQ_TABLE = {
+    'YS-JAN': 'Y',
     'AS-JAN': 'Y',
-    'YS': 'Y',
+    'AS': 'Y',
     '2QS-JAN': 'S',
     'QS-JAN': 'Q',
-    'min': 'MIN',
+    '12H': 'H12',
+    '6H': 'H6',
+    '3H': 'H3',
```

Comment:
These seems redundant since line 60-61 would just update those keys as well, and resolve to the same values.

Reply:
> These seems redundant since line 60-61 would just update those keys as well, and resolve to the same values.

Yes, you're right. I was weighting this option. :) However, I estimated the cost of the redundancy as insignificantly low, and I see value in having that mapping explicit, because it maps from older versions of pandas, while the `_TS_FREQ_TABLE` uses current pandas version, and if at some point we have to update the current version, we may forget to update the `_PANDAS_FREQ_TABLE` mapping, since the dependency may not be captured in a review. But I hesitated and I'm open to remove it, if you prefer. :)

Reply:
If I understand this PR correctly, we are trying to make this package compatible with both  pandas major versions 1.5.0 and 2.2.x. Then some things to consider:
- Breaking changes between the two versions need special handling from the code. This requires more effort to develop/maintain.
- Every test cases in CI need to work with both versions.

Instead, I would suggest we make this package compatible with 2.2.x only in this PR, this means as well updating the `requirements.txt` file for pandas version, and release a major version afterwards.

Reply:
I think the tests that are in this form are only compatible with 2.2.x